File: Easy/sasd/meta_sasddefaultview/meta_sasddefaultview.cs
/*
    Easy
    Copyright (C) 2020 Università degli Studi di Catania (www.unict.it)
    This program is free software: you can redistribute it and/or modify
    it under the terms of the GNU General Public License as published by
    the Free Software Foundation, either version 3 of the License, or
    (at your option) any later version.
    This program is distributed in the hope that it will be useful,
    but WITHOUT ANY WARRANTY; without even the implied warranty of
    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
    GNU General Public License for more details.
    You should have received a copy of the GNU General Public License
    along with this program.  If not, see <http://www.gnu.org/licenses/>.
*/

﻿using System;
using System.Data;
using metadatalibrary;
using metaeasylibrary;
//$CustomUsing$


namespace meta_sasddefaultview
{
    public class Meta_sasddefaultview : Meta_easydata 
	{
        public Meta_sasddefaultview(DataAccess Conn, MetaDataDispatcher Dispatcher) :
            base(Conn, Dispatcher, "sasddefaultview") {
				Name = "Settori artistico-scientifico disciplinari";
			EditTypes.Add("default");
			ListingTypes.Add("default");
			//$EditTypes$
        }

		private string[] mykey = new string[] {"idsasd"};

		public override string[] primaryKey() {
			return mykey;
		}

		//$SetDefault$

		//$Get_New_Row$

		//$IsValid$

		public override void DescribeColumns(DataTable T, string ListingType) {
			base.DescribeColumns(T, ListingType);

			foreach (DataColumn C in T.Columns) {
				DescribeAColumn(T, C.ColumnName, "", -1);
			}
			int nPos = 1;
					
			switch (ListingType) {
				case "default": {
						DescribeAColumn(T, "idsasd", "Codice", nPos++);
						DescribeAColumn(T, "sasd_codice", "Codice", nPos++);
						DescribeAColumn(T, "title", "Settore Artistico Scientifico Disciplinare", nPos++);
						DescribeAColumn(T, "areadidattica_title", "Area o ambito disciplinare", nPos++);
						DescribeAColumn(T, "sasd_codice_old", "Codice legge precedente", nPos++);
						break;
					}
					//$DescribeAColumn$
			}
		}

		public override string GetSorting(string ListingType) {
			switch (ListingType) {
				case "default": {
						return "sasd_codice asc , title desc";
					}
				//$GetSorting$
			}
			return base.GetSorting(ListingType);
		}

		//$GetStaticFilter$
		
		//$CustomCode$
    }
}

File: Easy/Portale/client/app_segreterie/metadata/sasd/meta_sasddefaultview.js
(function() {

    var MetaData = window.appMeta.MetaSegreterieData;

    function meta_sasddefaultview() {
        MetaData.apply(this, ["sasddefaultview"]);
        this.name = 'meta_sasddefaultview';
    }

    meta_sasddefaultview.prototype = _.extend(
        new MetaData(),
        {
            constructor: meta_sasddefaultview,
			superClass: MetaData.prototype,

			describeColumns: function (table, listType) {
				var nPos=1;
				var self = this;
				_.forEach(table.columns, function (c) {
					self.describeAColumn(table, c.name, '', null, -1, null);
				});
				switch (listType) {
					default:
						return this.superClass.describeColumns(table, listType);
					case 'default':
						this.describeAColumn(table, 'codice', 'Codice', null, 2000, 50);
						this.describeAColumn(table, 'sasd_title', 'Denominazione', null, 3000, 255);
						this.describeAColumn(table, 'areadidattica_title', 'Area o ambito disciplinare', null, 4200, 100);
						this.describeAColumn(table, 'sasd_codice_old', 'Codice legge precedente', null, 5000, 255);
						this.describeAColumn(table, 'sasd_tipoente', 'Tipologia di ente', null, 6000, null);
//$objCalcFieldConfig_default$
						break;
//$objCalcFieldConfig$
				}
				return appMeta.Deferred("describeColumns").resolve();
			},


			//$setCaptions$

			primaryKey: function () {
				return ["idsasd"];
			},


			//$isValidFunction$

			//$getStaticFilter$

			getSorting: function (listType) {
				switch (listType) {
					case "default": {
						return "codice asc , sasd_title desc";
					}
					//$getSortingin$
				}
				return this.superClass.getSorting(listType);
			}

        });

    window.appMeta.addMeta('sasddefaultview', new meta_sasddefaultview('sasddefaultview'));

	}());
